### PR TITLE
core: no retry on dropped PickResult

### PR DESCRIPTION
**Pull request by dapengzhang0**
Address #3553 for normal retry. Not covering the unimplemented hedging case.

### CODE REVIEW

File: core/src/test/java/io/grpc/internal/FailingClientStreamTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.junit.runners.JUnit4;
+
+/**
+ * Unit tests for {@link FailingClientStream}.
+ */
+@RunWith(JUnit4.class)
+public class FailingClientStreamTest {
+
+  @Test
+  public void rpcProgressPopulatedToListener() {
```
Comment (zhangkun83) [inline core/src/test/java/io/grpc/internal/FailingClientStreamTest.java]: Better to split it into two methods.

File: core/src/test/java/io/grpc/internal/RetriableStreamTest.java [hunk 2]
```diff
@@ -1506,6 +1508,30 @@ public void normalRetry_thenNoTransparentRetry_andNoMoreRetry() {
...
+    doReturn(mockStream1).when(retriableStreamRecorder).newSubstream(0);
+    retriableStream.start(masterListener);
+
+    verify(retriableStreamRecorder).newSubstream(0);
+    ArgumentCaptor<ClientStreamListener> sublistenerCaptor1 =
+        ArgumentCaptor.forClass(ClientStreamListener.class);
+    verify(mockStream1).start(sublistenerCaptor1.capture());
+
+    // normal retry
+    doReturn(mockStream2).when(retriableStreamRecorder).newSubstream(1);
```
Comment (zhangkun83) [inline core/src/test/java/io/grpc/internal/RetriableStreamTest.java]: All stubbing should happen at the beginning of the test. You don't know whether the stubbed method was called before this line.

File: core/src/test/java/io/grpc/internal/FailingClientStreamTest.java (hunk 1 above)
Comment (dapengzhang0) [reply core/src/test/java/io/grpc/internal/FailingClientStreamTest.java]: Done.

File: core/src/test/java/io/grpc/internal/RetriableStreamTest.java (hunk 2 above)
Comment (dapengzhang0) [reply core/src/test/java/io/grpc/internal/RetriableStreamTest.java]: Done.

**Comment by dapengzhang0** [comment]
@zhangkun83 Thanks for the review.

**Pull request merged**